FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/JobManager.cs

```diff
@@ -4,188 +4,52 @@
-
-            if (triggerStart >= _jobConfiguration.EndTime)
-            {
-                _logger.LogInformation($"The job trigger start time {triggerStart} is greater than the end time {_jobConfiguration.EndTime} in configuration, no need to start a new job.");
-                return null;
-            }
-
-            // End data period for this trigger
-            DateTimeOffset triggerEnd = GetTriggerEndTime();
-
-            if (triggerStart >= triggerEnd)
-            {
-                _logger.LogError("The start time '{triggerStart}' to trigger is in the future.", triggerStart);
-                throw new StartJobFailedException($"The start time '{triggerStart}' to trigger is in the future.");
-            }
-
-            var typeFilters = _typeFilterParser.CreateTypeFilters(
-                _filterConfiguration.FilterScope,
-                _filterConfiguration.RequiredTypes,
-                _filterConfiguration.TypeFilters);
-
-            var processedPatients = schedulerSetting?.ProcessedPatients;
-
-            var filterInfo =
-                new FilterInfo(_filterConfiguration.FilterScope, _filterConfiguration.GroupId, _jobConfiguration.StartTime, typeFilters, processedPatients);
-            var newJob = Job.Create(
-                _jobConfiguration.ContainerName,
-                JobStatus.New,
-                new DataPeriod(triggerStart, triggerEnd),
-                filterInfo);
-            return newJob;
-        }
-
-        private DateTimeOffset GetTriggerStartTime(SchedulerMetadata schedulerSetting)
-        {
-            var lastScheduledTo = schedulerSetting?.LastScheduledTimestamp;
-            return lastScheduledTo ?? _jobConfiguration.StartTime;
-        }
+            _logger.LogInformation("Job manager starts running.");
 
-        // Job end time could be null (which means runs forever) or a timestamp in the future like 2120/01/01.
-        // In this case, we will create a job to run with end time earlier that current timestamp.
-        // Also, FHIR data use processing time as lastUpdated timestamp, there might be some latency when saving to data store.
-        // Here we add a JobEndTimeLatencyInMinutes latency to avoid data missing due to latency in creation.
-        private DateTimeOffset GetTriggerEndTime()
-        {
-            // Add two minutes latency here to allow latency in saving resources to database.
-            var nowEnd = DateTimeOffset.UtcNow.AddMinutes(-1 * AzureBlobJobConstants.JobQueryLatencyInMinutes);
-            if (_jobConfiguration.EndTime != null
-                && nowEnd > _jobConfiguration.EndTime)
-            {
-                return _jobConfiguration.EndTime.Value;
-            }
-            else
-            {
-                return nowEnd;
-            }
-        }
+            using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
If I'm remembering it right, the `using` term will take effect just after the invoking the execution of this method, which means the cancellation token source will be disposed before the `await _jobHosting.StartAsync((byte)_jobConfiguration.QueueType, Environment.MachineName, cancellationTokenSource);` really taking effect.

Reply:
I will investigate it.

Reply:
Got the wrong clue before, by offline discussion I understand this function is fine to be so.

Reply:
Per our offline discussion, the `cancellationTokenSource` will be disposed after `jobHosting.StartAsync()` and `scheduler service` are completed as `await` is used. So it is ok for this scenario.